### PR TITLE
Add notebook using TensorFlow to perform the predictions

### PR DESCRIPTION
**Pull request by volpatto**
In this PR, a notebook using TensorFlow to perform the predictions through Neural Networks is provided (synced with a related .py script).

The notebook has 4 main parts:

* Data preprocessing: loading, splitting, feature engineering, etc.
* Hyperparameter (Adam's learning rate) tuning: a 4-folds Randomized Search is employed through `sklearn`.
* Train/Test stage: using the optimal learning rate, train/test two NNs: one with GHS constraints and other unconstrained.
* Results: several plots assessing the outcomes are provided. Results are checked for each std. property individually. A thermodynamical consistency check is done at the end of the notebook.

[Click here](https://github.com/ThermoPhase-FCSRG/stdprops-ml/blob/6aac7d2c42eaa0ec84039d40ce1a12aaeccf6a5b/notebooks/consistent-stdprops-ml-predictions-tensorflow.ipynb) to see the rendered notebook.

### CODE REVIEW

**Comment by volpatto** [comment]
- [ ] Fix all deps versions in the devenv file
